omas/omas_plot.py

```diff
@@ -845,28 +845,40 @@ def pf_active_overlay(ods, ax=None, **kw):
     mask = kw.pop('mask', numpy.ones(nc, bool))
     scalex, scaley = kw.pop('scalex', True), kw.pop('scaley', True)
 
+    def path_rectangle(rectangle):
+        """
+        :param rectangle: ODS sub-folder: element.*.geometry.rectangle
+        :return: n x 2 array giving the path around the outline of the coil element, suitable for input to Polygon()
+        """
+        x = rectangle['r']
+        y = rectangle['z']
+        dx = rectangle['width']
+        dy = rectangle['height']
+        return numpy.array([
+            [x - dx / 2., x - dx / 2., x + dx / 2., x + dx / 2.],
+            [y - dy / 2., y + dy / 2., y + dy / 2., y - dy / 2.]]).T
+
+    def path_outline(outline):
+        """
+        :param outline: ODS sub-folder: element.*.geometry.outline
+        :return: n x 2 array giving the path around the outline of the coil element, suitable for input to Polygon()
+        """
+        return numpy.array([outline['r'], outline['z']]).T
+
+    geo_type_map = ['outline', 'rectangle', 'oblique', 'arcs_of_circle']
```

Comment:
@orso82 I defined this in three places (once in OMFIT and twice in OMAS), and that's a problem. I want to define it centrally in OMAS (and import into OMFIT from OMAS) and make it dependent on IMAS version, since I anticipate it changing. What's the best way?

Reply:
@eldond define it as a function in `omas.omas_plot.py`, and have OMFIT importing it from omas in `OMFIT-source/classes/omfit_omas.py`. When you do that, place the import in a try statement, in case someone is running an older version of omas. Something like this:

```python
try:
   from omas.omas_plot import path_rectangle, path_outline
except ImportError:
   printe('You are running an old version of OMAS. Please update it.')
```

Reply:
@orso82 Okay, I can do that, but shouldn't it be more general than just being in plot? Someone else might need to know how to translate the integer geometry type into the string field name. Maybe it could go in omas_utils?